processor/redactionbykeyprocessor/processor_test.go

```diff
@@ -0,0 +1,563 @@
+	config := &Config{AllowedKeys: []string{"id", "name", "group"}}
+	allowed := map[string]pcommon.Value{
+		"id": pcommon.NewValueInt(5),
+	}
+	masked := map[string]pcommon.Value{
+		"name": pcommon.NewValueString("placeholder 4111111111111111"),
+	}
+
+	_, _, next := runTest(t, allowed, nil, masked, config)
+
+	firstOutILS := next.AllTraces()[0].ResourceSpans().At(0).ScopeSpans().At(0)
+	attr := firstOutILS.Spans().At(0).Attributes()
+	_, ok := attr.Get(redactedKeys)
+	assert.False(t, ok)
+	_, ok = attr.Get(redactedKeyCount)
+	assert.False(t, ok)
+	_, ok = attr.Get(maskedValues)
+	assert.False(t, ok)
+	_, ok = attr.Get(maskedValueCount)
+	assert.False(t, ok)
+}
+
+// TestMultipleBlockValues validates that the processor can block multiple
+// patterns
+func TestMultipleBlockValues(t *testing.T) {
+	config := &Config{AllowedKeys: []string{"id", "name", "mystery"},
+		BlockedValues: []string{"4[0-9]{12}(?:[0-9]{3})?", "(5[1-5][0-9]{3})"},
+		Summary:       "debug"}
+	allowed := map[string]pcommon.Value{
+		"id":      pcommon.NewValueInt(5),
+		"mystery": pcommon.NewValueString("mystery 52000"),
+	}
+	masked := map[string]pcommon.Value{
+		"name": pcommon.NewValueString("placeholder 4111111111111111"),
+	}
+	redacted := map[string]pcommon.Value{
+		"credit_card": pcommon.NewValueString("4111111111111111"),
+	}
+
+	_, _, next := runTest(t, allowed, redacted, masked, config)
+
+	firstOutILS := next.AllTraces()[0].ResourceSpans().At(0).ScopeSpans().At(0)
+	attr := firstOutILS.Spans().At(0).Attributes()
+	var deleted []string
+	for k := range redacted {
+		_, ok := attr.Get(k)
+		assert.False(t, ok)
+		deleted = append(deleted, k)
+	}
+	maskedKeys, ok := attr.Get(redactedKeys)
+	assert.True(t, ok)
+	assert.Equal(t, strings.Join(deleted, ","), maskedKeys.StringVal())
+	maskedKeyCount, ok := attr.Get(redactedKeyCount)
+	assert.True(t, ok)
+	assert.Equal(t, int64(len(deleted)), maskedKeyCount.IntVal())
+
+	blockedKeys := []string{"name", "mystery"}
+	maskedValues, ok := attr.Get(maskedValues)
+	assert.True(t, ok)
+	sort.Strings(blockedKeys)
+	assert.Equal(t, strings.Join(blockedKeys, ","), maskedValues.StringVal())
+	maskedValues.Equal(pcommon.NewValueString(strings.Join(blockedKeys, ",")))
+	maskedValueCount, ok := attr.Get(maskedValueCount)
+	assert.True(t, ok)
+	assert.Equal(t, int64(len(blockedKeys)), maskedValueCount.IntVal())
+	nameValue, _ := attr.Get("name")
+	mysteryValue, _ := attr.Get("mystery")
+	assert.Equal(t, "placeholder ****", nameValue.StringVal())
+	assert.Equal(t, "mystery ****", mysteryValue.StringVal())
+}
+
+// TestBlockValuesByKey validates that the processor can block by key
+func TestBlockValuesByKey(t *testing.T) {
+	config := &Config{
+		AllowAllKeys:       true,
+		BlockedValuesByKey: []BlockedValueByKey{{Key: "http.host", Regex: ".*:.*(@)"}},
+		Summary:            "debug"}
+	allowed := map[string]pcommon.Value{}
+	masked := map[string]pcommon.Value{
+		"http.host": pcommon.NewValueString("user:pass@domain.com/path"),
```

Comment:
`/path` would not be part of the `http.host` key, but this is a test data issue, the test itself is fine